lib/radiator_web/live/outline_component.ex

```diff
@@ -154,128 +144,107 @@ defmodule RadiatorWeb.OutlineComponent do
     |> reply(:noreply)
   end
 
+  def handle_event("save", %{"uuid" => uuid, "content" => content}, socket) do
+    user_id = socket.assigns.user_id
+    Dispatch.change_node_content(uuid, content, user_id, generate_event_id(socket.id))
+
+    socket
+    |> reply(:noreply)
+  end
+
   def handle_event(
-        "keydown",
-        %{"key" => "Enter", "uuid" => uuid, "value" => value, "selection" => selection},
+        "new",
+        %{"uuid" => uuid, "content" => content, "selection" => selection},
         socket
       ) do
-    {first, _} = String.split_at(value, selection["start"])
-    {_, last} = String.split_at(value, selection["end"])
+    {first, _} = String.split_at(content, selection["start"])
+    {_, last} = String.split_at(content, selection["end"])
 
     user_id = socket.assigns.user_id
     Dispatch.change_node_content(uuid, first, user_id, generate_event_id(socket.id))
 
     episode_id = socket.assigns.episode_id
+    params = %{"prev_id" => uuid, "content" => last, "episode_id" => episode_id}
+    Dispatch.insert_node(params, user_id, generate_event_id(socket.id))
+
+    socket
+    |> reply(:noreply)
+  end
 
-    params = %{
-      "prev_id" => uuid,
-      "content" => last,
-      "episode_id" => episode_id
-    }
+  def handle_event("merge_prev", %{"uuid" => uuid, "content" => content}, socket) do
+    prev_node = Outline.get_node_above(uuid)
 
-    Dispatch.insert_node(params, user_id, generate_event_id(socket.id))
+    if prev_node do
+      user_id = socket.assigns.user_id
+
+      Dispatch.change_node_content(
+        prev_node.uuid,
+        "#{prev_node.content}#{content}",
+        user_id,
+        generate_event_id(socket.id)
+      )
+
+      Dispatch.delete_node(uuid, user_id, generate_event_id(socket.id))
+    end
 
     socket
+    # |> stream_delete_by_dom_id(:nodes, "nodes-form-#{uuid}")
     |> reply(:noreply)
   end
 
-  def handle_event("keydown", %{"key" => key, "uuid" => uuid, "value" => ""}, socket)
-      when key in ["Backspace", "Delete"] do
-    user_id = socket.assigns.user_id
-    Dispatch.delete_node(uuid, user_id, generate_event_id(socket.id))
+  def handle_event("merge_next", %{"uuid" => uuid, "content" => content}, socket) do
+    next_node = Outline.get_node_below(uuid)
+
+    if next_node do
+      user_id = socket.assigns.user_id
+
+      Dispatch.change_node_content(
+        uuid,
+        "#{content}#{next_node.content}",
+        user_id,
+        generate_event_id(socket.id)
+      )
+
+      Dispatch.delete_node(next_node.uuid, user_id, generate_event_id(socket.id))
+    end
 
     socket
-    |> stream_delete_by_dom_id(:nodes, "nodes-form-#{uuid}")
+    # |> stream_delete_by_dom_id(:nodes, "nodes-form-#{uuid}")
     |> reply(:noreply)
```

Comment:
delete ?!

Reply:
✅ absolutely correct. code is removed now.